src/oasis-indexer/api.ts

```diff
@@ -1,38 +1,61 @@
 /** @file Wrappers around generated API */
 
-import axios, { AxiosResponse } from 'axios'
+import axios, { AxiosRequestConfig, AxiosResponse } from 'axios'
 import { paraTimesConfig } from '../config'
 import * as generated from './generated/api'
 import BigNumber from 'bignumber.js'
 import { UseQueryOptions } from '@tanstack/react-query'
 import { Layer } from './generated/api'
 import { getEthAccountAddress } from '../app/utils/helpers'
+import { Network } from '../types/network'
 
 export * from './generated/api'
 export type { RuntimeEvmBalance as Token } from './generated/api'
 
 declare module './generated/api' {
   export interface Transaction {
+    network: Network
     layer: Layer
   }
   export interface RuntimeTransaction {
+    network: Network
     layer: Layer
   }
   export interface Block {
+    network: Network
     layer: Layer
   }
   export interface RuntimeBlock {
+    network: Network
     layer: Layer
   }
   export interface Account {
+    network: Network
     layer: Layer
   }
   export interface RuntimeAccount {
+    network: Network
     layer: Layer
     address_eth?: string
   }
 }
 
+function wrapWithNetwork<P extends Array<any>, Q>(f: (...args: P) => Q): (net: Network, ...args: P) => Q {
+  return (net, ...args) => {
+    console.log('Should use', net)
+    return f(...args)
+  }
+}
+
+const baseUrls: Partial<Record<Network, string>> = {
+  testnet: process.env.REACT_TESTNET_APP_API,
+}
```

Comment:
No, actually these are only the overrides. The default one (for mainnet) is already specified elsewhere. I will rename the const to `baseUrlOverrides`.

Reply:
Renamed, added comment.

Reply:
Why not have all of them in one place?

Reply:
We have ```Axios.defaults.baseURL = process.env.REACT_APP_API``` elsewhere. This takes care of all mainnet calls.
If we removed this, we would have to add the extra option for _every_ other call, too, even for the ones we don't have manual overrides for.

So I would rather not do that.

Still, we could _repeat_ the mainnet URL here in this map, but then it would be somewhat misleading for those who read the code, and we would also pass unnecessary parameters to Axios. Still, we can do that if you would prefer to.

Reply:
OK, I have moved them in to the same location.

Reply:
Oh yeah, I do think we need overrides for every call. Pages should become explicit about using mainnet where they do, instead of defaulting to it. Not having separate testnet domain necessarily adds a bunch of this complexity

Reply:
> Oh yeah, I do think we need overrides for every call. Pages should become explicit about using mainnet where they do, instead of defaulting to it. 

That is exactly how it's implemented now.
The network name is always right there in the URL. 

(Except for global search, which search impartially on all networks.)

In the app code, we _never_ default to mainnet.

I have now removed the default URL from the code, so this is also true on the API level.

The bigger challenge is to make sure that all our API calls is prepared to handle deployments; ie. they demand a network parameter, override the query param, and add the option for the query override.

Actually, this sounds like something orval should do for us...